lib/msal-node/src/client/ManagedIdentityApplication.ts

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    AccessTokenEntity,
+    AuthenticationResult,
+    Authority,
+    AuthorityOptions,
+    CacheManager,
+    ClientAuthErrorCodes,
+    Constants,
+    CredentialFilter,
+    CredentialType,
+    DEFAULT_CRYPTO_IMPLEMENTATION,
+    DefaultStorageClass,
+    INetworkModule,
+    Logger,
+    NetworkManager,
+    ProtocolMode,
+    ResponseHandler,
+    ScopeSet,
+    TimeUtils,
+    createClientAuthError,
+} from "@azure/msal-common";
+import { ServiceFabric } from "./ManagedIdentitySources/ServiceFabric";
+import { AppService } from "./ManagedIdentitySources/AppService";
+import { CloudShell } from "./ManagedIdentitySources/CloudShell";
+import { AzureArc } from "./ManagedIdentitySources/AzureArc";
+import { Imds } from "./ManagedIdentitySources/Imds";
+import {
+    ManagedIdentityConfiguration,
+    ManagedIdentityNodeConfiguration,
+    buildManagedIdentityConfiguration,
+} from "../config/Configuration";
+import { version, name } from "../packageMetadata.js";
+import { ManagedIdentityRequest } from "../request/ManagedIdentityRequest";
+import { ManagedIdentityId } from "../config/ManagedIdentityId";
+import { CryptoProvider } from "../crypto/CryptoProvider";
+
+/**
+ * Class to initialize a managed identity and identify the service
+ */
+export class ManagedIdentityApplication {
+    private managedIdentityId: ManagedIdentityId;
+
+    private config: ManagedIdentityNodeConfiguration;
+
+    private logger: Logger;
+    private cacheManager: CacheManager;
+    private networkClient: INetworkModule;
+    private networkManager: NetworkManager;
+    private cryptoProvider: CryptoProvider;
+
+    private identitySource:
+        | ServiceFabric
+        | AppService
+        | CloudShell
+        | AzureArc
+        | Imds;
+
+    private fakeAuthority: Authority;
+
+    constructor(
```

Comment:
For consistency with `ConfidentialClientApplication`, we should have only 1 config param here, not 2.

Reply:
Making this change would mean editing the MSI config and making managedIdentityId a required input by the developer. I can do that if you'd like.

Reply:
I think that's a better developer experience? Managed Identity ID is mandatory.